src/tools/perf/lib/libperf.c

```diff
@@ -1696,14 +1696,17 @@ ucx_perf_do_warmup(ucx_perf_context_t *perf, const ucx_perf_params_t *params)
         /* Stop when reaching the deadline */
         stop_status = (ucs_get_time() > deadline) ? UCS_OK : UCS_INPROGRESS;
 
-        if (params->thread_count == 1) {
-            status = ucx_perf_test_exchange_status(perf, stop_status);
-        } else {
-#pragma omp barrier
-#pragma omp single copyprivate(status)
+#if _OPENMP
+        if (params->thread_count != 1) {
```

Comment:
can we remove `if (params->thread_count != 1`? If we have only one thread, the sequence barrier/single copyprivate/barrier would still be correct. do you see be perf issue there?